arcaflow_plugin_stressng/stressng_plugin.py

```diff
@@ -133,14 +113,14 @@ def stressng_run(
         os.remove(stressng_jobfile[1])
 
     return "success", WorkloadResults(
-        params,
-        system_un,
-        vminfo_un,
-        mmapinfo_un,
-        cpuinfo_un,
-        matrixinfo_un,
-        mqinfo_un,
-        hddinfo_un,
+        test_config=params,
+        systeminfo=system_un,
+        vminfo=results.get(Stressors.VM),
+        mmapinfo=results.get(Stressors.MMAP),
+        cpuinfo=results.get(Stressors.CPU),
+        matrixinfo=results.get(Stressors.MATRIX),
+        mqinfo=results.get(Stressors.MQ),
+        hddinfo=results.get(Stressors.HDD),
     )
```

Comment:
Given that the table approach improves the plugin maintenance above, removing explicit variable definitions, is it possible here in the success return to also use a dynamic approach so that we can get rid of all of the explicit references in this PR?

Reply:
I would love to, but I'm not sure how.

The call to the `WorkloadResults` c'tor previously used positional parameters, and (apparently) we want the omitted ones to be filled in with `None`.  So, we have to generate the parameter list somehow, with all the right values in the right order.  And, the obvious approaches to that tend to create an implicit coupling between the call site and the object definition, which I think leads to unnecessarily fragile code.

There's a good alternative to that which is to pass the parameters via keywords (by dictionary unpacking).  However, there is no particular relationship between the parameter names (e.g., `mqinfo`) and the stressor names (i.e., `"mq"`, the value of `Stressors.MQ`).  So, we would have to invent another table which mapped between them, and then use that to build a third table which is the argument dict (unless we can rename the fields in `WorkloadResults`...).  I had originally considered that, but concluded it was too much complication to ask Patricia to take into her PR.  However, if that sounds good to you, @dustinblack, particularly if we have the power to rename the fields in `WorkloadResults`, I'm happy to take that on (although, it's going to make Patricia's merge somewhat more "interesting"...).